### PR TITLE
Add C++ doc links to `library_design.md` and minor documentation fixes

### PR DESCRIPTION
**Pull request by isVoid**
## Description
This PR fixes some render issue with `library_design.md`, adds links to c++ documentation (maybe changed after merging [673](https://github.com/rapidsai/cuspatial/pull/673))

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: docs/source/api_docs/trajectory.rst [hunk 1]
```diff
@@ -7,5 +7,4 @@ Functions for identifying and grouping trajectories from point data.
 
 .. autofunction:: cuspatial.derive_trajectories
 .. autofunction:: cuspatial.trajectory_distances_and_speeds
-.. autofunction:: cuspatial.directed_hausdorff_distance
```
Comment (isVoid) [inline docs/source/api_docs/trajectory.rst:10]: This is a duplicate entry also present in `spatial.rst`

**Review by thomcom — APPROVED** [review]

**Review by harrism — APPROVED** [review]
Just suggest tightening up the wording of one sentence. Otherwise approve.

File: docs/source/developer_guide/library_design.md [hunk 2]
```diff
@@ -12,21 +13,23 @@ At a high level, `cuspatial` has three parts:
...
 [structure of arrays](https://en.wikipedia.org/wiki/Parallel_array) (SoA)
-format. Specifically, cuspatial adopts geoarrow format. Geoarrow is derived
-from the Apache Arrow list type, and it adopts a
-[`Variable-size List Layout`](https://arrow.apache.org/docs/format/Columnar.html#variable-size-list-layout),
-with the inner-most layer storing the points in a `Fixed-size list layout` array
-with `size==2`. 
+format. Specifically, cuspatial adopts geoarrow format. Geoarrow is an extension
+to arrow format. It adopts arrow's 
+[`Variable-size List Layout`](https://arrow.apache.org/docs/format/Columnar.html#variable-size-list-layout)
+to provide support to geometry arrays.
```
[the remark this answers is not in the document]
Comment (isVoid) [reply docs/source/developer_guide/library_design.md:32]: It is already linked here: https://github.com/rapidsai/cuspatial/blob/9c61ccc6bc99dec14e7bb33165c71d0eca600ac8/docs/source/developer_guide/library_design.md?plain=1#L44-L45

**Comment by thomcom** [comment]
rerun tests

**Comment by isVoid** [comment]
rerun tests

**Comment by isVoid** [comment]
@gpucibot merge

**Pull request merged**